lib/fizzy/types.hpp

```diff
@@ -336,6 +340,22 @@ struct Code
     // The decoded instructions' immediate values.
     // These are instruction-type dependent fixed size value in the order of instructions.
     bytes immediates;
+
+    Code() noexcept = default;
```

Comment:
I think you could get rid of default constructor by changing `parse_expr` to accumulate instructions and immediates in separate local variables and then constructing `Code` in the end with them.

Not sure it's worth the effort, but it seems that default-constructed `Code` is not used to represent any meaningful concept, but is just a convenience hack.

Reply:
I can actually remove these constructors.